From d1739bfb5bccde40e9809b97c46782d3bb86dd60 Mon Sep 17 00:00:00 2001
From: Jiani Liu <quic_jianil@quicinc.com>
Date: Thu, 6 Apr 2023 16:21:05 +0800
Subject: [PATCH] ath9k: update btcoex slot timing

Set WL_ACTIVE_TIME and WL_QC_TIME to 0, avoid WiFi MAC sending
ACK or CTS in coex period.

Also change first_slot_time from 0x1d to 0x12, as the BLE active
lead time is set to 20us.

Signed-off-by: Miaoqing Pan <miaoqing@codeaurora.org>
Signed-off-by: Jiani Liu <quic_jianil@quicinc.com>

diff --git a/drivers/net/wireless/ath/ath9k/btcoex.c b/drivers/net/wireless/ath/ath9k/btcoex.c
index 618c9df..b07b68f 100644
--- a/drivers/net/wireless/ath/ath9k/btcoex.c
+++ b/drivers/net/wireless/ath/ath9k/btcoex.c
@@ -69,8 +69,8 @@ void ath9k_hw_init_btcoex_hw(struct ath_hw *ah, int qnum)
 		.bt_priority_time = 2,
 		.bt_first_slot_time = 5,
 		.bt_hold_rx_clear = true,
-		.wl_active_time = 0x20,
-		.wl_qc_time = 0x20,
+		.wl_active_time = 0x0,
+		.wl_qc_time = 0x0,
 	};
 	bool rxclear_polarity = ath_bt_config.bt_rxclear_polarity;
 	u8 time_extend = ath_bt_config.bt_time_extend;
@@ -80,7 +80,7 @@ void ath9k_hw_init_btcoex_hw(struct ath_hw *ah, int qnum)
 		rxclear_polarity = !ath_bt_config.bt_rxclear_polarity;
 
 	if (AR_SREV_SOC(ah)) {
-		first_slot_time = 0x1d;
+		first_slot_time = 0x12;
 		time_extend = 0xa;
 
 		btcoex_hw->bt_coex_mode3 =
-- 
2.7.4

